src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts

```diff
@@ -1875,6 +1871,9 @@ export abstract class AbstractTaskService extends Disposable implements ITaskSer
 			}
 		}
 		this._setRecentlyUsedTask(executeResult.task);
+		if (this._configurationService.getValue(TaskSettingId.Reconnection) === true && runSource !== TaskRunSource.Reconnect && executeResult.task.type !== '$customized') {
```

Comment:
Why not `_setPersistentTask` for `$customized` tasks?

Reply:
Because they seem to get resolved into a task that isn't of type $customized eventually. When they're of type $customized, the call to getRecentlyUsedKey throws because the key for them is a mapKey IE string so throws when parse is called